modules/external_petsc_solver/test/tests/partition/tests

```diff
@@ -0,0 +1,56 @@
+[Tests]
+  design = "ExternalProblem.md"
+  issues = "#16729"
+
+  [2D_2]
+    type = 'Exodiff'
+    input = 'moose_as_master.i'
+    exodiff = 'moose_as_master_out.e moose_as_master_out_sub_app0.e'
+    cli_args = 'Outputs/hide="picard_its"'
+    requirement = "MOOSE shall be able to match partitions from DMDA as much as possible using 2 cores."
```

Comment:
Let's be consistent about requirements. You use "the system" in your other tests spec. I like that better than "MOOSE".